actor/src/main/scala/org/apache/pekko/pattern/RetrySupport.scala

```diff
@@ -172,30 +362,34 @@ object RetrySupport extends RetrySupport {
       }
     }
 
-    require(maxAttempts >= 0, "Parameter maxAttempts must >= 0.")
+    def doRetry(nextAttempt: Int): Future[T] = delayFunction(nextAttempt) match {
+      case Some(delay) =>
+        if (delay.length < 1)
+          retry(attempt, shouldRetry, maxAttempts, delayFunction, nextAttempt)
+        else
+          after(delay, scheduler) {
+            retry(attempt, shouldRetry, maxAttempts, delayFunction, nextAttempt)
+          }
+      case None =>
+        retry(attempt, shouldRetry, maxAttempts, delayFunction, nextAttempt)
+      case null =>
+        Future.failed(new IllegalArgumentException("The delayFunction of retry should not return null."))
+    }
+
     require(attempt != null, "Parameter attempt should not be null.")
+    require(maxAttempts >= 0, "Parameter maxAttempts must >= 0.")
```

Comment:
We should add these necessary checks in a unified place, and I think RetrySupport L120 has already achieved this. (Alternatively, we can remove it there.)

Reply:
Thanks, let me do some clean up then.